### DISCUSSION TITLE
Streamed Resources

### DISCUSSION THREAD

**Discussion by crazynds** [laravel/framework#50242] [Ideas]
I was taking a look at pull request #49873, and I identified a feature that would be very useful in the APIResources that already exist.

I generally do the following in my controllers:
```php
public function index(IndexRequest $request)
    {
        $page = $request->input('page', 1);
        $qtd = $request->input('qtd', 20);
        $query = Client::with(['user', 'address']);

        return ClientResource::collection($query->paginate($qtd));
    }
```
And in some entities it may take time to start transmitting data to the user.
I believe it is plausible to add the StreamedResources functionality to the API Resources, especially in the static ::collection() function. I don't think it's a difficult addition.
 
What do you think of the idea? I can work on a sketch and maybe make a pull request if it's interesting.

**Comment by ghost** [comment]


**Reply by crazynds** [reply]
This comment is unrelated to this topic.

**Reply by ghost** [reply]
Ok. I removed it and posted a new idea.
